### PR TITLE
 webView: render realm emoji.

### PR DESCRIPTION
**Pull request by jainkuniya**
#1623 

@borisyankov ready for review.
Let me know changes :)

### CODE REVIEW

**Comment by borisyankov** [comment]
You need to handle realm emoji outside the reaction too.
I think you are using a bit too much custom code for that.
Go over the diff now and consider what you can simplify.

**Comment by jainkuniya** [comment]
> You need to handle realm emoji outside the reaction too.

Inline realm emojis in messages are automatically handled as they already contain img tag. 
Am I missing out something? 🤔

File: src/render-html/es3.js [hunk 1]
```diff
@@ -170,12 +170,12 @@ documentBody.addEventListener('click', function (e) {
     e.preventDefault();
   }
 
-  if (e.target.matches('.reaction')) {
+  if (e.target.matches('.reaction') || e.target.matches('.realm-reaction')) {
```
Comment (borisyankov) [inline src/render-html/es3.js]: Make sure to render both emojis similarly as to not require `or`s and `if`s.

File: src/render-html/es3.js [hunk 2]
```diff
@@ -170,12 +170,12 @@ documentBody.addEventListener('click', function (e) {
...
     sendMessage({
       type: 'reaction',
-      name: e.target.getAttribute('data-name'),
+      name: e.target.getAttribute('data-name') || e.target.parentNode.getAttribute('data-name'),
       messageId: +getMessageIdFromNode(e.target),
-      voted: e.target.classList.contains('self-voted')
+      voted: e.target.classList.contains('self-voted') || e.target.parentNode.classList.contains('self-voted')
```
Comment (borisyankov) [inline src/render-html/es3.js]: Same here.

File: src/render-html/js.js [hunk 3]
```diff
@@ -179,12 +179,14 @@ documentBody.addEventListener('click', e => {
     e.preventDefault();
   }
 
-  if (e.target.matches('.reaction')) {
+  if (e.target.matches('.reaction') || e.target.matches('.realm-reaction')) {
     sendMessage({
       type: 'reaction',
-      name: e.target.getAttribute('data-name'),
+      name: e.target.getAttribute('data-name') || e.target.parentNode.getAttribute('data-name'),
```
Comment (borisyankov) [inline src/render-html/js.js]: And likely here.

File: src/render-html/js.js [hunk 4]
```diff
@@ -179,12 +179,14 @@ documentBody.addEventListener('click', e => {
...
       messageId: +getMessageIdFromNode(e.target),
-      voted: e.target.classList.contains('self-voted'),
+      voted:
+        e.target.classList.contains('self-voted') ||
```
Comment (borisyankov) [inline src/render-html/js.js]: ... and here.

File: src/render-html/messageAsHtml.js [hunk 5]
```diff
@@ -10,17 +10,27 @@ const messageDiv = (id, msgClass, flags) =>
      ${flags.map(flag => `data-${flag}="true" `).join('')}
     >`;
 
-const briefMessageAsHtml = ({ id, content, flags, timeEdited, isOutbox, reactions, ownEmail }) => `
+const briefMessageAsHtml = ({
+  auth,
```
Comment (borisyankov) [inline src/render-html/messageAsHtml.js]: You can preprocess the data passed in a selector, so you don't need to pass `auth`.

File: src/render-html/messageReactionAsHtml.js [hunk 6]
```diff
@@ -1,8 +1,30 @@
...
-`;
+const getRealmEmojiHtml = (
+  reaction: AggregateReaction,
+  realmEmoji: RealmEmojiType,
+  realm: string,
+) =>
+  `<img class="realm-reaction" src="${getFullUrl(
+    realmEmoji.source_url,
+    realm,
+  )}" height="auto" width="16"/>
```
Comment (borisyankov) [inline src/render-html/messageReactionAsHtml.js]: Do not add `height` and `width` as attributes but via `css.js`

File: src/types.js [hunk 7]
```diff
@@ -508,3 +508,9 @@ export type PresenceAggregated = {
   status: string,
   timestamp: number,
 };
+
+export type AggregateReaction = {
```
Comment (borisyankov) [inline src/types.js]: Just call it `ReactionType`?

**Comment by borisyankov** [comment]
Also, make sure the images are aligned correctly vertically.

**Comment by borisyankov** [comment]
Good to merge. Please rebase.

**Comment by jainkuniya** [comment]
Rebased 👍

**Pull request merged**